### PR TITLE
hal: nuvoton: m2l31: support SPI 3-wire mode

### PR DESCRIPTION
**Pull request by cyliangtw**
This PR is to update m2l31/spi.h to support SPI 3-wire mode enable and disable control.

### CODE REVIEW

File: m2l31x/StdDriver/inc/spi.h [hunk 1]
```diff
@@ -115,6 +115,24 @@ extern "C"
...
 #define SPI_CLR_UNIT_TRANS_INT_FLAG(spi)   ((spi)->STATUS = SPI_STATUS_UNITIF_Msk)
 
+/**
+  * @brief      Disable Slave 3-wire mode.
+  * @param[in]  spi The pointer of the specified SPI module.
+  * @return     None.
+  * @details    Clear SLV3WIRE bit of SPI_SSCTL register to disable Slave 3-wire mode.
+  * \hideinitializer
+  */
+#define SPI_DISABLE_3WIRE_MODE(spi)   ( (spi)->SSCTL &= ~SPI_SSCTL_SLV3WIRE_Msk )
```
Comment (fabiobaltieri) [inline m2l31x/StdDriver/inc/spi.h]:
```suggestion
#define SPI_DISABLE_3WIRE_MODE(spi)   ((spi)->SSCTL &= ~SPI_SSCTL_SLV3WIRE_Msk)
```

For coherency with the other definitions, same below

Comment (cyliangtw) [reply m2l31x/StdDriver/inc/spi.h]: yes, I modified the format for consistency.

Comment (fabiobaltieri) [reply m2l31x/StdDriver/inc/spi.h]: I still see them the same, sure you pushed the right thing?

Comment (cyliangtw) [reply m2l31x/StdDriver/inc/spi.h]: Sorry, miss to add this update before force push.
Now, fixed.

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**